src/ant/include/ant/AntennaChecker.hh

```diff
@@ -68,6 +68,48 @@ struct Violation
   int diode_count_per_gate;
 };
 
+// A struct that contains a net as well as one of its routing levels.
+// This struct is used to index the map, allowed_wire_length which is created 
+// in the class below.
+struct NetLayerPair
+{
+    dbNet* net;
+    int routing_level;
+ 
+    // default constructor
+    NetLayerPair() {}
+    // constructor
+    NetLayerPair(dbNet* net, int routing_level): net(net), routing_level(routing_level) {}
+
+    // overload `<` operator to use this struct as a key for an std::map
+    bool operator<(const NetLayerPair &ob) const {
+        int lhs_id = net->getId();
+        int rhs_id = ob.net->getId();
+        return std::tie(lhs_id, routing_level) < std::tie(rhs_id, ob.routing_level);
+
+    }
+};
+
+
+// A struct that contains 3 fields:
+//          1- net_layer: a NetLayerPair pointer
```

Comment:
It make more sense to comment the member vars in the class than add a lengthy header that repeats their names